crates/bevy_core_pipeline/src/motion_blur/node.rs

```diff
@@ -0,0 +1,116 @@
+use bevy_ecs::{query::QueryItem, world::World};
+use bevy_render::{
+    extract_component::ComponentUniforms,
+    globals::GlobalsBuffer,
+    render_graph::{NodeRunError, RenderGraphContext, ViewNode},
+    render_resource::{
+        BindGroupEntry, BindingResource, Operations, PipelineCache, RenderPassColorAttachment,
+        RenderPassDescriptor,
+    },
+    renderer::RenderContext,
+    view::{Msaa, ViewTarget},
+};
+
+use crate::prepass::ViewPrepassTextures;
+
+use super::{
+    pipeline::{MotionBlurPipeline, MotionBlurPipelineId},
+    MotionBlur,
+};
+
+#[derive(Default)]
+pub struct MotionBlurNode;
+
+impl ViewNode for MotionBlurNode {
+    type ViewData = (
+        &'static ViewTarget,
+        &'static MotionBlurPipelineId,
+        &'static ViewPrepassTextures,
+    );
+    fn run(
+        &self,
+        _graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        (view_target, pipeline_id, prepass_textures): QueryItem<Self::ViewData>,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let motion_blur_pipeline = world.resource::<MotionBlurPipeline>();
+        let pipeline_cache = world.resource::<PipelineCache>();
+        let settings_uniforms = world.resource::<ComponentUniforms<MotionBlur>>();
+        let Some(pipeline) = pipeline_cache.get_render_pipeline(pipeline_id.0) else {
+            return Ok(());
+        };
+
+        let Some(settings_binding) = settings_uniforms.uniforms().binding() else {
+            return Ok(());
+        };
+        let (Some(prepass_motion_vectors_texture), Some(prepass_depth_texture)) =
+            (&prepass_textures.motion_vectors, &prepass_textures.depth)
+        else {
+            return Ok(());
+        };
+        let Some(globals_uniforms) = world.resource::<GlobalsBuffer>().buffer.binding() else {
+            return Ok(());
+        };
+
+        let post_process = view_target.post_process_write();
+
+        let msaa = world.resource::<Msaa>();
+        let layout = if msaa.samples() == 1 {
+            &motion_blur_pipeline.layout
+        } else {
+            &motion_blur_pipeline.layout_msaa
+        };
+
+        let bind_group = render_context.render_device().create_bind_group(
```

Comment:
Use the new bind group API please.

Reply:
Can you provide docs or something? I have no idea what that is.

Reply:
The docs on `BindGroup` point me to `create_bind_group`, which Im already using.

Reply:
Updated main and copied one of the other pipelines